### PR TITLE
Business documents in PDF format rev1

### PR DESCRIPTION
**Pull request by SerhiiTrehubenko**
#555

### CODE REVIEW

**Review by belosh59 — APPROVED** [review]

File: goods-partner-backend/src/main/java/com/goodspartner/service/util/PdfFileCompiler.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+        if (pdfQualityDocumentUrls.isEmpty()) {
+            return;
+        }
+
+        for (String pdfQualityDocumentUrl : pdfQualityDocumentUrls) {
+            byte[] pdfQualityDocument = openSourceFile(pdfQualityDocumentUrl);
+
+            if (Objects.isNull(pdfQualityDocument)) {
+                continue;
```
Comment (TarasOrlovskyi) [inline goods-partner-backend/src/main/java/com/goodspartner/service/util/PdfFileCompiler.java:121]: interesting.. can you change it?

File: goods-partner-backend/src/main/java/com/goodspartner/service/dto/DocumentPdfType.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.goodspartner.service.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
+public enum DocumentPdfType {
+    PDF(new String[]{".pdf"}),
+    IMAGES(new String[]{".png", ".jpg", ".bmp"});
```
Comment (TarasOrlovskyi) [inline goods-partner-backend/src/main/java/com/goodspartner/service/dto/DocumentPdfType.java:10]: please, add .jpeg also

File: goods-partner-backend/src/main/java/com/goodspartner/service/util/PdfFileCompiler.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        Set<String> qualityDocImages = pdfDocumentDto.getQualityDocuments().get(DocumentPdfType.IMAGES);
+
+        if (qualityDocImages.isEmpty()) {
+            return;
+        }
+
+        for (String qualityDocImage : qualityDocImages) {
+            byte[] image = openSourceFile(qualityDocImage);
+            if (Objects.isNull(image)) {
+                continue;
```
Comment (TarasOrlovskyi) [inline goods-partner-backend/src/main/java/com/goodspartner/service/util/PdfFileCompiler.java:170]: interesting.. can you change it?

File: goods-partner-backend/src/main/java/com/goodspartner/service/util/PdfFileCompiler.java (hunk 1 above)
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/util/PdfFileCompiler.java:121]: add the log here please

File: goods-partner-backend/src/main/java/com/goodspartner/service/util/PdfFileCompiler.java (hunk 3 above)
Comment (belosh59) [inline goods-partner-backend/src/main/java/com/goodspartner/service/util/PdfFileCompiler.java:170]: please add log here

File: goods-partner-backend/src/main/java/com/goodspartner/service/util/PdfFileCompiler.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        }
+    }
+
+    private byte[] openSourceFileFromExternalHost(String fileUrl) throws IOException {
+        try (InputStream fileFromExternalHost = new URL(fileUrl).openStream()) {
+            Objects.requireNonNull(fileFromExternalHost);
+            return fileFromExternalHost.readAllBytes();
+        } catch (IOException e) {
+            log.error("Quality document {} does not exist at external host", fileUrl);
+            throw new IOException(e);
```
Comment (TarasOrlovskyi) [inline goods-partner-backend/src/main/java/com/goodspartner/service/util/PdfFileCompiler.java:193]: should throw custom Exception

File: goods-partner-backend/src/main/java/com/goodspartner/service/dto/DocumentPdfType.java (hunk 2 above)
Comment (SerhiiTrehubenko) [reply goods-partner-backend/src/main/java/com/goodspartner/service/dto/DocumentPdfType.java]: Added

File: goods-partner-backend/src/main/java/com/goodspartner/service/util/PdfFileCompiler.java (hunk 1 above)
Comment (SerhiiTrehubenko) [reply goods-partner-backend/src/main/java/com/goodspartner/service/util/PdfFileCompiler.java:121]: Log is present before "continue;"

**Review by belosh59 — APPROVED** [review]

**Pull request merged**